core/blockchain.go

```diff
@@ -72,14 +71,14 @@ var (
 	blockValidationTimer = metrics.NewRegisteredTimer("chain/validation", nil)
 	blockExecutionTimer  = metrics.NewRegisteredTimer("chain/execution", nil)
 	blockWriteTimer      = metrics.NewRegisteredTimer("chain/write", nil)
-	blockReorgAddMeter   = metrics.NewRegisteredMeter("chain/reorg/drop", nil)
-	blockReorgDropMeter  = metrics.NewRegisteredMeter("chain/reorg/add", nil)
 
 	blockPrefetchExecuteTimer   = metrics.NewRegisteredTimer("chain/prefetch/executes", nil)
 	blockPrefetchInterruptMeter = metrics.NewRegisteredMeter("chain/prefetch/interrupts", nil)
 
 	errInsertionInterrupted = errors.New("insertion is interrupted")
 	errCommitmentNotFound   = errors.New("randomness commitment not found")
+	errParentNotCanonical   = errors.New("parent not canonical, reorgs disabled")
+	errAlreadyCanonical     = errors.New("already a canonical block in same level, reorgs disabled")
```

Comment:
I would suggest not using error values such as this, using error values tends to lead to having errors devoid of useful information.

In the case of errParentNotCanonical, what would be useful, would be to know the hashes and numbers of the expected canonical parent and the actual canonical head block. If you see this popping up in the logs that information will massively narrow down the potential scope of the problem.

The only argument for using error values is if we wanted to check for them explicitly, but these are never checked. Also there are problems with explicitly checking error values, this [article](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully) explains the issues around using error values in depth and suggests better alternatives.

In this case I just removing these values and returning an error that contains relevant contextual information would be fine.

Reply:
Agree. Changing it with your other recommendation. Thanks!